### ISSUE TITLE
Equivalent of hyperopt's normal/lognormal parameters

### ISSUE THREAD

**Issue by liori** [optuna/optuna#1200]
Is it possible to have a parameter taken from a normal/lognormal distribution? Effectively a parameter either not constrained at all, or constrained only to positive numbers. Thanks!

**Comment by toshihikoyanase** [comment]
I'm sorry for the delayed response.

> Is it possible to have a parameter taken from a normal/lognormal distribution?

I think the normal distribution has the mean and standard deviation to specify the values to be suggested. Currently, Optuna does not support them. Since this is the first request for such distributions,  could you share your usage of such distributions?

**Comment by liori** [comment]
> Since this is the first request for such distributions, could you share your usage of such distributions?

Sure! The story is that I was optimizing hyperparameters for training neural networks. I used `hyperopt` and set the learning rate as a lognormal with mode of `0.01` and the interquartile range was around `0.1` to `0.001`. After a night of computations I woke up to see that `hyperopt`'s TPE found an optimum around `10`, much better than anything close to my initial bet!

Out of curiosity then I wanted to try optuna on the same problem, but found that there were no unbounded distributions available. And if I were to use a uniform or loguniform distribution, I'd never think a learning rate that high would be a good value to try.

**Comment by toshihikoyanase** [comment]
@liori Thank you for sharing your use case.

> I used hyperopt and set the learning rate as a lognormal with mode of 0.01 and the interquartile range was around 0.1 to 0.001. After a night of computations I woke up to see that hyperopt's TPE found an optimum around 10, much better than anything close to my initial bet!

Interesting! That's an advantage of optimization without range constraints. 

> Out of curiosity then I wanted to try optuna on the same problem, but found that there were no unbounded distributions available.

Although Optuna does not support unbounded distributions, I think we can utilize Optuna's visualization feature to revise the range of distribution. For instance, [Parallel Coordinates plot](https://optuna.readthedocs.io/en/stable/reference/visualization.html#optuna.visualization.plot_parallel_coordinate) and [Slice plot](https://optuna.readthedocs.io/en/stable/reference/visualization.html#optuna.visualization.plot_slice) shows the relationship between the objective values and the parameter values. In your case, I think you could see the positive correlation between the objective value and the learning rate in those plots and you would expand the range.

Regarding the support of unbounded distributions, we need careful API design because of the following reason:

1.  *Name of distribution and suggest API*: We received several feedbacks that claimed `suggest_uniform` did not sample values from uniform distributions. Now we are redesigning the interface (e.g., https://github.com/optuna/optuna/issues/510), and want to introduce new distributions carefully.
1. *Sampling algorithms*: I think `RandomSampler`, `TPESampler` and `CmaEsSampler` can support unbounded distributions, but I'm not sure for the other samplers. If we support unbounded distributions, users may not change samplers casually.

So, I'd like to start small (e.g., creating a toy example of a specific sampler) if I implement unbounded distributions.

Thank you again for your feedback.
